src/main/scala/interfaces/Verification.scala

```diff
@@ -21,25 +21,22 @@ import viper.silicon.state.terms.Term
 
 /* TODO: Make VerificationResult immutable */
 sealed abstract class VerificationResult {
-  var previous: Option[NonFatalResult] = None
+  var previous: Set[VerificationResult] = Set()
 
   def isFatal: Boolean
   def &&(other: => VerificationResult): VerificationResult
 
-  def allPrevious: List[VerificationResult] =
-    previous match {
-      case None => Nil
-      case Some(vr) => vr :: vr.allPrevious
-    }
-
-  def append(other: NonFatalResult): VerificationResult =
-    previous match {
-      case None =>
-        this.previous = Some(other)
+  def combine(other: => VerificationResult): VerificationResult = {
+    val r: VerificationResult = other
```

Comment:
Please add a comment that explains that the line here executes `other`. This is subtle and often overlooked, in particular by people not regularly working with Scala.

Reply:
Added this now.

Reply:
Added this now

Reply:
Was that committed? I can't see it here.

Reply:
I think it somehow still shows the old diff here, but it's in [this commit](https://github.com/tschoesi/silicon/commit/a19df8b0c6684de33cd35eda37429c705c82f6be).